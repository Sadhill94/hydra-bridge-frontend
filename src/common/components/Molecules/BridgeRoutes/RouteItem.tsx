import React, { ReactNode } from "react";
import styled from "styled-components";

import Icon from "../../Icon/Icon";
import { ContainerCard } from "../../Atoms/Containers/Container";
import {
  StyledBridgeArrow,
  StyledBridgeNetwork,
  StyledBridgeRoute,
  StyledBridgeRouteAmount,
} from "./styles";
import { FakeButton } from "../../Atoms/Buttons/Button";

import { IconKeys } from "../../../commonTypes";
import { stakenetTheme as theme } from "../../../../shell/theme/stakenetTheme";
import { devicesUp } from "../../../../media";

const CustomFakeButton = styled(FakeButton)<{
  hasBoxShadow?: boolean;
}>`
  &:focus,
  &:focus-within {
    background-color: ${theme.colors.blue.darker};
  }
`;
export const RouteItemContainerCard = styled(ContainerCard)<{
  isSelected: boolean;
}>`
  min-height: 10rem;
  padding: 1rem;
  background-color: ${theme.colors.blue.darkest};
  ${(props) =>
    props.isSelected ? `background-color: ${theme.colors.blue.darker}` : ""};
  border-radius: ${theme.borderRadius.lg};
  position: relative;
  z-index: 3;

  &:hover,
  &:focus {
    background-color: ${theme.colors.blue.darker};
  }
  @media only screen and ${devicesUp.md} {
    padding: 1.4rem 2.4rem;
  }
`;

export type RouteItemProps = {
  coinSymbol: IconKeys;
  bridgeSymbol: IconKeys;
  amountIn: string | number;
  amountOut: string | number;
  bridgeDisplayName: string;
  routeId: number;
  isSelected: boolean;
  onRouteSelect: (id: number) => void;
  children?: ReactNode;
};

const RouteItem = ({
  coinSymbol,
  bridgeSymbol,
  amountIn,
  amountOut,
  bridgeDisplayName,
  routeId,
  isSelected,
  onRouteSelect,
  children,
}: RouteItemProps) => {
  return (
    <CustomFakeButton
      ariaLabel={"select route"}
      hasBoxShadow={true}
      onClick={() => onRouteSelect(routeId)}
    >
      <RouteItemContainerCard isSelected={isSelected}>
        <StyledBridgeRoute>
          <StyledBridgeRouteAmount>
            <Icon
              name={coinSymbol}
              width={"2.8rem"}
              height={"3.2rem"}
              className={"amount__icon"}
            />
            <span className={"amount__number"}>{amountIn}</span>
          </StyledBridgeRouteAmount>
          <StyledBridgeArrow>
            <Icon
              name={"doubleArrowRight"}
              width={"2.1rem"}
              height={"1.8rem"}
              className={"arrow__icon"}
            />
          </StyledBridgeArrow>
          <StyledBridgeNetwork>
            <Icon
              name={bridgeSymbol}
              width={"2.6rem"}
              height={"2.3rem"}
              className={"network__icon"}
            />
            <p className={"network__name"}>{bridgeDisplayName}</p>
          </StyledBridgeNetwork>
          <StyledBridgeArrow>
            <Icon
              name={"doubleArrowRight"}
              width={"2.1rem"}
              height={"1.8rem"}
              className={"arrow__icon"}
            />
          </StyledBridgeArrow>
          <StyledBridgeRouteAmount rtl={true}>
            <Icon
              name={coinSymbol}
              width={"2.8rem"}
              height={"3.2rem"}
              className={"amount__icon"}
            />
            <span className={"amount__number"}>{amountOut}</span>
          </StyledBridgeRouteAmount>
        </StyledBridgeRoute>
        {children}
      </RouteItemContainerCard>
    </CustomFakeButton>
  );
};

export default RouteItem;
