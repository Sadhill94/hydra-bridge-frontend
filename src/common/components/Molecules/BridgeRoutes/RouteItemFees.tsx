import React from "react";
import { useTranslation } from "react-i18next";
import styled from "styled-components";

import { ContainerCard } from "../../Atoms/Containers/Container";

import { stakenetTheme as theme } from "../../../../shell/theme/stakenetTheme";

const StyledText = styled.p`
  color: ${theme.colors.white};
  font-size: ${theme.paragraph.xs};
  margin: 0;
  min-width: 15rem;
  padding: 0.5rem 1.2rem;
`;

export type RouteItemFeesProps = {
  transactionCostInUsd: number;
  serviceTime: number;
};
const RouteItemFees = ({
  transactionCostInUsd,
  serviceTime,
}: RouteItemFeesProps) => {
  const { t } = useTranslation();

  const getGasFee = () =>
    // eg: round(0.39979797399225586 * 100) = 40 / 100 = 0.4 to fixed(2) = 0.40
    (Math.round(transactionCostInUsd * 100) / 100).toFixed(2);

  const getServiceTimeOutput = () => `~ ${serviceTime / 60} min`;
  return (
    <ContainerCard
      borderRadius={theme.borderRadius.lg}
      padding={"0"}
      bg={theme.colors.gray["medium-dark"]}
      boxShadow={theme.boxShadow.sm}
    >
      <StyledText data-testid="route-fees">
        {t("gas-fee")} : ~ {getGasFee()} $ | {getServiceTimeOutput()}
      </StyledText>
    </ContainerCard>
  );
};

export default RouteItemFees;
