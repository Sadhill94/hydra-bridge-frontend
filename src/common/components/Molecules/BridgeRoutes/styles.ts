import { ReactNode } from "react";
import styled from "styled-components";

import { ContainerCard } from "../../Atoms/Containers/Container";

import { stakenetTheme as theme } from "../../../../shell/theme/stakenetTheme";
import { devicesUp } from "../../../../media";

export const RouteItemContainerCard = styled(ContainerCard)<{
  isSelected: boolean;
  hasChildren?: boolean;
  hasError?: boolean;
}>`
  min-height: ${(props) =>
    props.hasChildren ? "10rem" : props.hasError ? "10rem" : "auto"};
  padding: 1rem;
  background-color: ${(props) =>
    props.isSelected ? theme.colors.gray.dark : theme.colors.blue.darkest};
  box-shadow: ${theme.boxShadow.sm};
  border-radius: ${theme.borderRadius.lg};

  &:hover,
  &:focus {
    background-color: ${theme.colors.gray.dark};
  }
  @media only screen and ${devicesUp.md} {
    padding: 1.8rem 1.8rem 0.8rem 1.8rem !important;
  }
`;

export const StyledBridgeRoute = styled.div<{ children?: ReactNode }>`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;
  width: 100%;
  margin-bottom: ${(props) => (props.children ? theme.margin.md : 0)};

  @media only screen and ${devicesUp.sm} {
    align-items: center;
  }
`;

export const StyledBridgeRouteAmount = styled.div<{ rtl?: boolean }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: ${theme.paragraph.md};
  color: ${theme.colors.white};
  min-width: 25%;
  max-width: 25%;

  .amount {
    &__number {
      word-break: break-word;
      flex-grow: 1;
    }

    &__icon {
      margin-bottom: ${theme.margin.sm};
      transform: scale(0.8);
    }
  }

  @media only screen and ${devicesUp.sm} {
    .amount {
      &__icon {
        margin-bottom: ${theme.margin.md};
        transform: scale(1);
      }
    }
  }

  @media only screen and ${devicesUp.md} {
    flex-direction: ${(props) => (props.rtl ? "row-reverse" : "row")};
    align-items: center;

    .amount {
      &__number {
        font-size: ${theme.paragraph.lg};
        text-align: ${(props) => (props.rtl ? "right" : "left")};
        letter-spacing: 0.1rem;
      }
      &__icon {
        margin: 0 ${(props) => (props.rtl ? 0 : theme.margin.md)} 0
          ${(props) => (props.rtl ? theme.margin.md : 0)};
      }
    }
  }
`;

export const StyledBridgeArrow = styled.div`
  text-align: center;
  margin-bottom: ${theme.margin.sm};
  min-width: 10%;
  max-width: 10%;

  .arrow__icon {
    transform: scale(0.8);
  }
  @media only screen and ${devicesUp.sm} {
    margin-bottom: ${theme.margin.md};

    .arrow__icon {
      transform: scale(1);
    }
  }
  @media only screen and ${devicesUp.md} {
    margin-bottom: 0;
  }
`;

export const StyledBridgeChain = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 30%;
  max-width: 30%;

  .chain {
    &__name {
      font-size: ${theme.paragraph.md};
      color: ${theme.colors.white};
      margin: 0;
      display: flex;
      flex-grow: 1;
      text-align: center;
    }
    &__icon {
      margin-bottom: ${theme.margin.sm};
      text-align: center;
    }
    &__group {
      display: none;
    }
  }

  @media only screen and ${devicesUp.sm} {
    .chain {
      &__icon {
        min-width: 2.6rem;
      }
    }
  }

  @media only screen and ${devicesUp.md} {
    flex-direction: row;
    justify-content: center;
    text-align: center;
    width: 100%;
    .chain {
      &__name {
        display: none;
        width: 100%;
        font-size: ${theme.paragraph.lg};
        letter-spacing: 0.1rem;
      }
      &__icon {
        margin: 0 ${theme.margin.sm} 0 0;
        &--sm {
          display: none;
        }
      }

      &__group {
        display: flex;
        justify-content: center;
        align-items: center;
        flex-wrap: wrap;
        flex-grow: 1;
        font-size: ${theme.paragraph.lg};
        color: ${theme.colors.white};
        span {
          margin-top: 0.3rem;
        }
      }
    }
  }
`;
