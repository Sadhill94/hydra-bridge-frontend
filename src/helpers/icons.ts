import {
  HOP,
  HOP_BRIDGE,
  HOP_BRIDGE_GOERLI,
  POLYGON,
  POLYGON_BRIDGE,
  POLYGON_BRIDGE_GOERLI,
} from "../common/constants";
import { IconKeys } from "../common/commonTypes";

/**
 * Helper to allow to match the bridge name.
 * Later on used to get the the icon key
 * @param bridgeName
 */
export const getBridgeIconName = (bridgeName: string): string => {
  if (bridgeName === POLYGON_BRIDGE || bridgeName === POLYGON_BRIDGE_GOERLI) {
    return POLYGON;
  }

  if (bridgeName === HOP_BRIDGE || bridgeName === HOP_BRIDGE_GOERLI) {
    return HOP;
  }

  return "";
};

/**
 * Helper to easily get the icon key from a bridge
 * @param bridgeName
 */
export const getBridgeIcon = (bridgeName: string): IconKeys =>
  `${bridgeName.toLocaleLowerCase()}Bridge` as IconKeys;

/**
 * Helper to easily get the icon key from a coin
 * @param coinSymbol
 */
export const getCoinIcon = (coinSymbol: string): IconKeys =>
  `${coinSymbol.toLocaleLowerCase()}Coin` as IconKeys;
