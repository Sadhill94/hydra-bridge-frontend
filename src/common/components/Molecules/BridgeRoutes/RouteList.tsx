import React, { ReactNode } from "react";
import { useTranslation } from "react-i18next";
import styled from "styled-components";

import RouteItem from "./RouteItem";
import RouteItemFees from "./RouteItemFees";

import { RouteDto } from "../../../dtos";
import { getBridgeIcon, getCoinIcon } from "../../../../helpers/icons";
import { stakenetTheme as theme } from "../../../../shell/theme/stakenetTheme";
import { devicesUp } from "../../../../media";

const StyledDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 10rem;
  width: 100%;
  font-size: ${theme.paragraph.md};
  text-align: center;
  color: ${theme.colors.black};

  @media only screen and ${devicesUp.md} {
    font-size: ${theme.paragraph.lg};
  }
`;

export type BridgeRouteListProps = {
  routes: RouteDto[];
  selectedRouteId?: number;
  onRouteSelect: (routeId: number) => void;
};

const RouteList = ({
  routes,
  selectedRouteId,
  onRouteSelect,
}: BridgeRouteListProps) => {
  const { t } = useTranslation();

  const renderRouteItems = () => {
    try {
      const routeList: ReactNode = routes.map((route: RouteDto) => {
        const {
          id,
          transactionCoastUsd,
          bridgeRoute: {
            amountIn,
            amountOut,
            bridgeName,
            fromAsset: { symbol },
            bridgeInfo: { displayName, serviceTime },
          },
        } = route;
        const coinSymbol = getCoinIcon(symbol);
        const bridgeSymbol = getBridgeIcon(bridgeName);

        return (
          <div
            key={id}
            style={{ width: "100%", marginBottom: theme.margin.md }}
          >
            <RouteItem
              coinSymbol={coinSymbol}
              bridgeSymbol={bridgeSymbol}
              amountIn={amountIn}
              amountOut={amountOut}
              bridgeDisplayName={displayName}
              routeId={id}
              isSelected={selectedRouteId === id}
              onRouteSelect={onRouteSelect}
            >
              <RouteItemFees
                transactionCostInUsd={transactionCoastUsd}
                serviceTime={serviceTime}
              />
            </RouteItem>
          </div>
        );
      });
      return routeList;
    } catch (err) {
      return (
        <StyledDiv style={{ height: "10rem" }}>
          {t("error-showing-routes")}
        </StyledDiv>
      );
    }
  };

  return <>{renderRouteItems()}</>;
};

export default RouteList;
