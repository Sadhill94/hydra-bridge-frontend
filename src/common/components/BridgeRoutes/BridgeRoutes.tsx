import styled from "styled-components";
import { RouteDto } from "../../dtos";
import { getFlexStart } from "../../styles";
import RouteList from "../Molecules/BridgeRouteList/RouteList";
import { RectangleSkeleton } from "../Atoms/Skelletons/styles";

const Root = styled.div`
  margin-top: 10px;
  background: rgb(255, 255, 255);
  box-shadow: rgb(0 0 0 / 1%) 0 0 1px, rgb(0 0 0 / 4%) 0 4px 8px,
    rgb(0 0 0 / 4%) 0 16px 24px, rgb(0 0 0 / 1%) 0 24px 32px;
  border-radius: 24px;
  margin-left: auto;
  margin-right: auto;
  z-index: 1;
  padding: 10px;
`;

const TitleContainer = styled.div`
  ${getFlexStart};
  margin-top: 10px;
  margin-bottom: 10px;
`;
const Title = styled.div`
  font-size: ${({ theme }) => theme.heading.xs};
  font-weight: 700;
`;

type Props = {
  inProgress: boolean;
  routes: RouteDto[];
  selectedRouteId?: number;
  onRouteSelect: (routeId: number) => void;
};
const BridgeRoutes = ({
  routes,
  inProgress,
  selectedRouteId,
  onRouteSelect,
}: Props) => {
  return (
    <Root>
      <TitleContainer>
        <Title>Available routes:</Title>
      </TitleContainer>
      {inProgress ? (
        <RectangleSkeleton height={"10rem"} />
      ) : (
        <RouteList
          routes={routes}
          selectedRouteId={selectedRouteId!}
          onRouteSelect={onRouteSelect}
        />
      )}
    </Root>
  );
};

export default BridgeRoutes;
