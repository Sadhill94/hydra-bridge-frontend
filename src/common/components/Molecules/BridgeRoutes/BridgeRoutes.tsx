import React, { ReactNode } from "react";
import { useTranslation } from "react-i18next";

import RouteList from "./RouteList";
import { RectangleSkeleton } from "../../Atoms/Skelletons/styles";
import { Container } from "../../Atoms/Containers/Container";
import { ContainerType } from "../../../enums";
import { InputLabel as Label } from "../../Atoms/Label/Label";
import { FlexWrapper } from "../../Atoms/Wrappers/Wrapper";
import Accordion from "../Accordion/Accordion";
import RoundedBubble from "../../Atoms/RoundedBubble/RoundedBubble";

import { RouteDto } from "../../../dtos";
import { AccordionHeader } from "../Accordion/AccordionHeaders";
import { AccordionContent } from "../Accordion/styles";

export type BridgeRoutesProps = {
  inProgress: boolean;
  routes: RouteDto[];
  selectedRouteId?: number;
  onRouteSelect: (routeId: number) => void;
};
const BridgeRoutes = ({
  routes,
  inProgress,
  selectedRouteId,
  onRouteSelect,
}: BridgeRoutesProps) => {
  const { t } = useTranslation();

  const renderHeader = (): ReactNode => (
    <AccordionHeader>
      <FlexWrapper
        flexDirection={"row"}
        alignItems={"center"}
        justifyContent={"space-between"}
      >
        <Label margin={"0"} style={{ width: "100%", textAlign: "left" }}>
          {t("available-routes")}
        </Label>
        <RoundedBubble>{routes.length || 0}</RoundedBubble>
      </FlexWrapper>
    </AccordionHeader>
  );
  const renderContent = (): ReactNode => {
    return inProgress ? (
      <AccordionContent>
        <RectangleSkeleton
          className={"rectangle-skeleton"}
          height={"10rem"}
          style={{ marginBottom: "1rem" }}
        />
        <RectangleSkeleton
          className={"rectangle-skeleton"}
          height={"10rem"}
          style={{ marginBottom: "1rem" }}
        />
        <RectangleSkeleton className={"rectangle-skeleton"} height={"10rem"} />
      </AccordionContent>
    ) : (
      <AccordionContent>
        <RouteList
          routes={routes}
          selectedRouteId={selectedRouteId!}
          onRouteSelect={onRouteSelect}
        />
      </AccordionContent>
    );
  };

  return (
    <Container type={ContainerType.XXXL} noGutter={true}>
      <Accordion header={renderHeader()} content={renderContent()} />
    </Container>
  );
};

export default BridgeRoutes;
