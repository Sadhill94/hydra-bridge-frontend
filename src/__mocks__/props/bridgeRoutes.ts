import { BridgeRouteItemProps } from "../../common/components/Molecules/BridgeRoutes/RouteItem";

export const routeItemBaseProps: BridgeRouteItemProps = {
  amountIn: "0.0001",
  amountOut: "0.0001",
  coinSymbol: "ethCoin",
  bridgeSymbol: "polygonBridge",
  bridgeDisplayName: "Polygon",
  routeId: 2,
  isSelected: false,
  onRouteSelect: (id: number) => alert(`on route select id: ${id}`),
};
