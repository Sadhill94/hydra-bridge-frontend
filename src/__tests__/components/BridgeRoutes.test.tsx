import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";

import BridgeRoutes, {
  BridgeRoutesProps,
} from "../../common/components/Molecules/BridgeRoutes/BridgeRoutes";

import { mockBridgeRoutesBaseProps } from "../../__mocks__/props/bridgeRoutes";
import { removeDynamicIdFromSnapshot } from "../helpers";

describe("The Bridge routes", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render correctly when not in progress and no error", async () => {
    const props: BridgeRoutesProps = mockBridgeRoutesBaseProps;

    const { container, asFragment } = render(<BridgeRoutes {...props} />);

    fireEvent.click(screen.getByRole("button"));
    const renderedRoutes =
      container.querySelectorAll<HTMLDivElement>(`[id^="route-"]`);

    expect(screen.getByText(/available-routes/)).toBeInTheDocument();
    expect(screen.getByText(props.routes.length)).toBeInTheDocument();
    expect(renderedRoutes.length).toBe(props.routes.length);
    expect(removeDynamicIdFromSnapshot(asFragment())).toMatchSnapshot();
  });
  it("should render the skeleton when in progress", async () => {
    const props: BridgeRoutesProps = {
      ...mockBridgeRoutesBaseProps,
      inProgress: true,
    };

    const { container, asFragment } = render(<BridgeRoutes {...props} />);

    fireEvent.click(screen.getByRole("button"));
    const renderedRoutes =
      container.querySelectorAll<HTMLDivElement>(`[id^="route-"]`);
    const skeletons = container.querySelectorAll<HTMLDivElement>(
      ".rectangle-skeleton"
    );

    expect(screen.getByText(/available-routes/)).toBeInTheDocument();
    expect(renderedRoutes.length).toBe(0);
    expect(skeletons.length).toBe(3);
    expect(removeDynamicIdFromSnapshot(asFragment())).toMatchSnapshot();
  });
});
