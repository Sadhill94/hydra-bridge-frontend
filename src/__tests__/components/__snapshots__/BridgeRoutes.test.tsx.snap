// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Bridge routes should render correctly when not in progress and no error 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL sc-eCImPb bXyKXZ czizOc"
    type="3xl"
  >
    <div
      class="sc-bdvvtL sc-jRQBWg fGXmcv lkItHs"
    >
      <div
        aria-expanded="true"
        aria-label="open accordion"
        class="sc-iqseJM eIsPzL"
        role="button"
        tabindex="0"
      >
        <div
          class="sc-fotOHu bnkEsk"
        >
          <div
            class="sc-fotOHu fbDECk"
          >
            <label
              class="sc-hBUSln dubAMH"
              style="width: 100%; text-align: left;"
            >
              available-routes
            </label>
            <div
              class="sc-dJjYzT jLGZDQ"
            >
              2
            </div>
          </div>
          <svg
            class="sc-ieecCq iNYKmN"
            color="#FFFFFF"
            height="1.7rem"
            isopen="1"
            style="margin-left: 1.7rem;"
            width="1.7rem"
          >
            cut-arrow-right.svg
          </svg>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="rah-animating rah-animating--down rah-animating--to-height-auto"
        style="width: 100%; height: 0px; overflow: hidden; transition: height 500ms ease 0ms;"
      >
        <div
          style=""
        >
          <div
            class="sc-bkkeKt gZVIEP"
          >
            <div
              id="route-8"
              style="width: 100%; margin-bottom: 2rem;"
            >
              <div
                aria-expanded="false"
                aria-label="select route"
                class="sc-iqseJM bzJIfm sc-crHmcD gYEbqi"
                role="button"
                tabindex="0"
              >
                <div
                  class="sc-bdvvtL sc-jRQBWg fGXmcv eIbvzN sc-egiyK gssaBE"
                >
                  <div
                    class="sc-gKclnd konHGI"
                  >
                    <div
                      class="sc-iCfMLu gYGgwA"
                    >
                      <svg
                        class="amount__icon"
                        color="#FFFFFF"
                        height="4rem"
                        width="4rem"
                      >
                        usdc.svg
                      </svg>
                      <span
                        class="amount__number"
                      >
                        1
                      </span>
                    </div>
                    <div
                      class="sc-furwcr fhgROL"
                    >
                      <svg
                        class="arrow__icon"
                        color="#FFFFFF"
                        height="2rem"
                        width="2rem"
                      >
                        double-arrow-right.svg
                      </svg>
                    </div>
                    <div
                      class="sc-pVTFL kOcWqs"
                    >
                      <svg
                        class="bridge__icon"
                        color="#FFFFFF"
                        height="4rem"
                        width="4rem"
                      >
                        hop.svg
                      </svg>
                      <p
                        class="bridge__name"
                      >
                        Hop Goerli
                      </p>
                    </div>
                    <div
                      class="sc-furwcr fhgROL"
                    >
                      <svg
                        class="arrow__icon"
                        color="#FFFFFF"
                        height="2rem"
                        width="2rem"
                      >
                        double-arrow-right.svg
                      </svg>
                    </div>
                    <div
                      class="sc-iCfMLu fRazBy"
                    >
                      <svg
                        class="amount__icon"
                        color="#FFFFFF"
                        height="4rem"
                        width="4rem"
                      >
                        usdc.svg
                      </svg>
                      <span
                        class="amount__number"
                      >
                        1.0
                      </span>
                    </div>
                  </div>
                  <div
                    class="sc-bdvvtL sc-jRQBWg fGXmcv dQJyMJ"
                  >
                    <p
                      class="sc-bqiRlB jIWwtk"
                      data-testid="route-fees"
                    >
                      gas-fee : ~ 0.61 $ | ~ 5 min
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              id="route-6"
              style="width: 100%; margin-bottom: 2rem;"
            >
              <div
                aria-expanded="false"
                aria-label="select route"
                class="sc-iqseJM bzJIfm sc-crHmcD gYEbqi"
                role="button"
                tabindex="0"
              >
                <div
                  class="sc-bdvvtL sc-jRQBWg fGXmcv eIbvzN sc-egiyK dKiIhB"
                >
                  <div
                    class="sc-gKclnd konHGI"
                  >
                    <div
                      class="sc-iCfMLu gYGgwA"
                    >
                      <svg
                        class="amount__icon"
                        color="#FFFFFF"
                        height="4rem"
                        width="4rem"
                      >
                        usdc.svg
                      </svg>
                      <span
                        class="amount__number"
                      >
                        1
                      </span>
                    </div>
                    <div
                      class="sc-furwcr fhgROL"
                    >
                      <svg
                        class="arrow__icon"
                        color="#FFFFFF"
                        height="2rem"
                        width="2rem"
                      >
                        double-arrow-right.svg
                      </svg>
                    </div>
                    <div
                      class="sc-pVTFL kOcWqs"
                    >
                      <svg
                        class="bridge__icon"
                        color="#FFFFFF"
                        height="4rem"
                        width="4rem"
                      >
                        polygon.svg
                      </svg>
                      <p
                        class="bridge__name"
                      >
                        Polygon POS Goerli
                      </p>
                    </div>
                    <div
                      class="sc-furwcr fhgROL"
                    >
                      <svg
                        class="arrow__icon"
                        color="#FFFFFF"
                        height="2rem"
                        width="2rem"
                      >
                        double-arrow-right.svg
                      </svg>
                    </div>
                    <div
                      class="sc-iCfMLu fRazBy"
                    >
                      <svg
                        class="amount__icon"
                        color="#FFFFFF"
                        height="4rem"
                        width="4rem"
                      >
                        usdc.svg
                      </svg>
                      <span
                        class="amount__number"
                      >
                        1.0
                      </span>
                    </div>
                  </div>
                  <div
                    class="sc-bdvvtL sc-jRQBWg fGXmcv dQJyMJ"
                  >
                    <p
                      class="sc-bqiRlB jIWwtk"
                      data-testid="route-fees"
                    >
                      gas-fee : ~ 0.63 $ | ~ 30 min
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`The Bridge routes should render the skeleton when in progress 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL sc-eCImPb bXyKXZ czizOc"
    type="3xl"
  >
    <div
      class="sc-bdvvtL sc-jRQBWg fGXmcv lkItHs"
    >
      <div
        aria-expanded="true"
        aria-label="open accordion"
        class="sc-iqseJM eIsPzL"
        role="button"
        tabindex="0"
      >
        <div
          class="sc-fotOHu bnkEsk"
        >
          <div
            class="sc-fotOHu fbDECk"
          >
            <label
              class="sc-hBUSln dubAMH"
              style="width: 100%; text-align: left;"
            >
              available-routes
            </label>
            <div
              class="sc-dJjYzT jLGZDQ"
            >
              2
            </div>
          </div>
          <svg
            class="sc-ieecCq iNYKmN"
            color="#FFFFFF"
            height="1.7rem"
            isopen="1"
            style="margin-left: 1.7rem;"
            width="1.7rem"
          >
            cut-arrow-right.svg
          </svg>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="rah-animating rah-animating--down rah-animating--to-height-auto"
        style="width: 100%; height: 0px; overflow: hidden; transition: height 500ms ease 0ms;"
      >
        <div
          style=""
        >
          <div
            class="sc-bkkeKt gZVIEP"
          >
            <div
              class="sc-ksdxgE jNGrHQ rectangle-skeleton"
              height="10rem"
              style="margin-bottom: 1rem;"
            />
            <div
              class="sc-ksdxgE jNGrHQ rectangle-skeleton"
              height="10rem"
              style="margin-bottom: 1rem;"
            />
            <div
              class="sc-ksdxgE jNGrHQ rectangle-skeleton"
              height="10rem"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
